From aad521ce3790d137ad74e9db8714f8cfba05a6ee Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 13 Oct 2020 20:11:20 +0300
Subject: [PATCH] [SW-12754] gaudi: security: disable all pcie rrs

When disabling the security - disable all PCIe IF RRs.
This is in preparation to using more PCIe IF RRs for security
implementation.
Since during hard reset we do not reset the PCIe IF, those RRs are not
reset and will keep protecting the regions.

Change-Id: I138c5ce50284a9baa0b3125b41a9b143ab7d0335
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index e093bdd..15d89ef 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1179,16 +1179,24 @@
 	preboot_secure_hbw_rr(7, 0x7FFBFF0000, 0x3FFFFFFF8000);
 }
 
+#define PCIE_WRAP_RRS_NUM	32
+
 static void preboot_security_pci_disable(void)
 {
-	u32 hit;
+	int i;
+	u32 hit, base, prot_base;
 
-	regmap_writel(0, mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MIN_0);
-	regmap_writel(0, mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MAX_0);
-	regmap_writel(0, mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MIN_0);
-	regmap_writel(0, mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MAX_0);
-	regmap_writel(0, mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MIN_1);
-	regmap_writel(0, mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MAX_1);
+	for (i = 0; i < PCIE_WRAP_RRS_NUM; i++) {
+		base = i * sizeof(u32);
+		prot_base = base + mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MIN_0;
+		regmap_writel(0, prot_base);
+		prot_base = base + mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MAX_0;
+		regmap_writel(0, prot_base);
+		prot_base = base + mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MIN_0;
+		regmap_writel(0, prot_base);
+		prot_base = base + mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MAX_0;
+		regmap_writel(0, prot_base);
+	}
 
 	/* PCIe RTR RR - HBM WP 0x0 - 0x07FFFFFFFF */
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_BASE_LOW_AW_0);
